frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Decoder.as

```diff
@@ -0,0 +1,258 @@
+    //--------------------------------------------------------------------------
+
+    /**
+     * Constructor.
+     *  
+     *  @langversion 3.0
+     *  @playerversion Flash 9
+     *  @playerversion AIR 1.1
+     *  @productversion Royale 0.9.4
+     */
+    COMPILE::SWF
+    public function Base64Decoder()
+    {
+        super();
+        data = new ByteArray();
+    }
+
+	COMPILE::JS
+    public function Base64Decoder()
+    {
+        super();
+    }
+
+
+    //--------------------------------------------------------------------------
+    //
+    //  Methods
+    //
+    //--------------------------------------------------------------------------
+
+    /**
+     * Decodes a Base64 encoded String and adds the result to an internal
+     * buffer. Strings must be in ASCII format. 
+     * 
+     * <p>Subsequent calls to this method add on to the internal
+     * buffer. After all data have been encoded, call <code>toByteArray()</code>
+     * to obtain a decoded <code>flash.utils.ByteArray</code>.</p>
+     * 
+     * @param encoded The Base64 encoded String to decode.
+     *  
+     *  @langversion 3.0
+     *  @playerversion Flash 9
+     *  @playerversion AIR 1.1
+     *  @productversion Royale 0.9.4
+     */
+
+    COMPILE::JS
+    public function decode(data:String):String
+    {
+
+		var bytes:Object = new Base64JSWrapper().toByteArray(data);
+		var decodedString:String = new TextEncoderLiteWrapper('utf-8').decode(bytes);
+
+		return decodedString;
+    }
+
+
+    COMPILE::SWF
+    public function decode(encoded:String):void
+    {
+        for (var i:uint = 0; i < encoded.length; ++i)
+        {
+            var c:Number = encoded.charCodeAt(i);
+
+            if (c == ESCAPE_CHAR_CODE)
+                work[count++] = -1;
+            else if (inverse[c] != 64)
+                work[count++] = inverse[c];
+            else
+                continue;
+
+            if (count == 4)
+            {
+                count = 0;
+                data.writeByte((work[0] << 2) | ((work[1] & 0xFF) >> 4));
+                filled++;
+
+                if (work[2] == -1)
+                    break;
+
+                data.writeByte((work[1] << 4) | ((work[2] & 0xFF) >> 2));
+                filled++;
+
+                if (work[3] == -1)
+                    break;
+
+                data.writeByte((work[2] << 6) | work[3]);
+                filled++;
+            }
+        }
+    }
+
+    COMPILE::SWF
+    public function drain():ByteArray
+    {
+        var result:ByteArray = new ByteArray();
+
+        var oldPosition:uint = data.position;    
+        data.position = 0;  // technically, shouldn't need to set this, but carrying over from previous implementation
+        result.writeBytes(data, 0, data.length);        
+        data.position = oldPosition;
+        result.position = 0;
+        
+        filled = 0;
+        return result;
+    }
+
+    COMPILE::SWF
+    public function flush():ByteArray
+    {
+        if (count > 0)
```

Comment:
If we are not using that condition - Remove it completely

Reply:
@piotrzarzycki21  After commenting out IResourceManager and ResourceManager - i am really not sur to handle - the condition became meaningless. Sorry, i missed that. Will fix it.

Reply:
I just noticed that I do not understand above comment fully. Are you saying that there can be scenario that you un-comment that code ? If NOT let's remove it instead commenting it and leave it in source code.

Reply:
I tought that i have to make the class can be compiled without error for swf and functional for JS. From your comments, i see that it is not a correct approach. I can remove flush method from the class instead of commenting but after, i should remove the functions calling flush and so on. Do you want me to go that direction ? Just to make it clear and do not let missunderstanding.